### PR TITLE
Button: Update hover & default states

### PR DESCRIPTION
**Pull request by dimitrisnl**
Change-type: minor
Signed-off-by: Dimitrios Lytras <dimitrios@balena.io>

---
##### Contributor checklist

- [ ] I have rebuilt the README with `npm run build:docs`
- [ ] I have regenerated screenshots for any affected components with `npm run generate-screenshots`
- [ ] I have regenerated jest snapshots for any affected components with `npm run jest -- -u`

##### Reviewer Guidelines
- When submitting a review, please pick:
  - '*Approve*' if this change would be acceptable in the codebase (even if there are minor or cosmetic tweaks that could be improved).
  - '*Request Changes*' if this change would not be acceptable in our codebase (e.g. bugs, changes that will make development harder in future, security/performance issues, etc).
  - '*Comment*' if you don't feel you have enough information to decide either way (e.g. if you have major questions, or you don't understand the context of the change sufficiently to fully review yourself, but want to make a comment)
---

### CODE REVIEW

**Comment by dimitrisnl** [comment]
There is no need to update the snapshots for now.

**Comment by dimitrisnl** [comment]
Screen Shot 2019-05-14 at 2 53 35 PM
Screen Shot 2019-05-14 at 2 53 43 PM
Screen Shot 2019-05-14 at 2 53 51 PM

**Comment by dimitrisnl** [comment]
May-14-2019 14-54-40

**Comment by dimitrisnl** [comment]
Updating the hover effect made the default buttons look _really_ of, so I had to update them too according to the styleguide.

File: src/components/Button.tsx [hunk 1]
```diff
@@ -92,7 +131,28 @@ const getStyledButton = (
 		);
 	}
 
-	if ((outline && !active) || !isPrimary) {
+	const themeColorVariants = Object.keys(originalProps.theme.colors);
+	const hasColorVariant = some(themeColorVariants, partial(has, originalProps));
```
Comment (sradevski) [inline src/components/Button.tsx]: Can't we just call `getColor` and see if any color was returned?

Comment (dimitrisnl) [reply src/components/Button.tsx]: Great point @sradevski - I overcomplicated it there

**Comment by dimitrisnl** [comment]
Surprised that the test still fail

**Pull request merged**